pkg/plugin/integration_test.go

```diff
@@ -351,27 +316,377 @@ func TestCreateThenUpdateCausesNoRotations(t *testing.T) {
+			change:         func(cs *api.OpenShiftManagedCluster) { cs.Config.Images.WebConsole = "newImage" },
+		},
+		{
+			name:           "change MasterServicePrincipalProfile",
+			expectRotation: map[rotationType]bool{rotationMaster: true, rotationInfra: false, rotationSync: false, rotationCompute: false},
+			change: func(cs *api.OpenShiftManagedCluster) {
+				cs.Properties.MasterServicePrincipalProfile.Secret = uuid.NewV4().String()
+			},
+		},
+		{
+			name:           "change WorkerServicePrincipalProfile",
+			expectRotation: map[rotationType]bool{rotationMaster: false, rotationInfra: true, rotationSync: false, rotationCompute: true},
+			change: func(cs *api.OpenShiftManagedCluster) {
+				cs.Properties.WorkerServicePrincipalProfile.Secret = uuid.NewV4().String()
+			},
+		},
+		{
+			name:           "change IdentityProfile",
+			expectRotation: map[rotationType]bool{rotationMaster: true, rotationInfra: false, rotationSync: true, rotationCompute: false},
+			change: func(cs *api.OpenShiftManagedCluster) {
+				provider := cs.Properties.AuthProfile.IdentityProviders[0].Provider.(*api.AADIdentityProvider)
+				provider.Secret = uuid.NewV4().String()
+			},
+		},
+		{
+			name:           "change CIDR",
+			expectRotation: map[rotationType]bool{rotationMaster: false, rotationInfra: false, rotationSync: false, rotationCompute: false},
+			change: func(cs *api.OpenShiftManagedCluster) {
+				cs.Properties.NetworkProfile.VnetCIDR = "11.0.0.0/8"
+				for i := range cs.Properties.AgentPoolProfiles {
+					cs.Properties.AgentPoolProfiles[i].SubnetCIDR = "11.0.0.0/24"
+				}
+			},
+		},
+	}
+
+	log := logrus.NewEntry(logrus.StandardLogger())
+	ctx := context.Background()
+	cs := newTestCs()
+	az := newFakeAzureCloud(log)
+	p, _, err := setupNewCluster(ctx, log, cs, az)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			log.Infof("--- Test: %s", tt.name)
+			beforeBlob, beforeSyncChecksum, err := getHashes(az, cs)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			tt.change(cs)
+
+			errs := p.Validate(ctx, cs, nil, false)
+			if errs != nil {
+				t.Fatal(errs)
+			}
+			perr := p.CreateOrUpdate(ctx, cs, true, getFakeDeployer(log, cs, az))
+			if perr != nil {
+				t.Fatal(perr)
+			}
+
+			afterBlob, afterSyncChecksum, err := getHashes(az, cs)
+			if err != nil {
+				t.Fatal(err)
+			}
+			rotations := getRotations(beforeBlob, afterBlob, beforeSyncChecksum, afterSyncChecksum)
+			if !reflect.DeepEqual(tt.expectRotation, rotations) {
+				t.Fatalf("rotation mismatch: expected %v, got %v", tt.expectRotation, rotations)
+			}
+		})
+	}
+}
+
+func TestHowActionsCauseRotations(t *testing.T) {
+	tests := []struct {
+		name           string
+		change         func(ctx context.Context, log *logrus.Entry, p *plugin, cs *api.OpenShiftManagedCluster, az *fakecloud.AzureCloud) error
```

Comment:
I wish there was some way we could avoid this long collection of posargs on all these lambdas, it's a bit hard to read :(

Reply:
Yeah, it's mostly the CreateOrUpdate() that takes a heap of args that are not defined at that time of defining the function here.. I could pull "log" out and create one above..